presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkRow.java

```diff
@@ -17,18 +17,44 @@
 
 import static java.util.Objects.requireNonNull;
 
-public class SerializedPrestoSparkPage
+public class PrestoSparkRow
         implements Serializable
 {
+    private static final int INSTANCE_SIZE = Long.BYTES * 2 /* headers */
+            + Integer.BYTES /* partition */
+            + Integer.BYTES /* length */
+            + Long.BYTES /* bytes pointer */
+            + Long.BYTES * 2 /* bytes headers */
+            + Integer.BYTES /* bytes length */;
+
+    private final int partition;
+    private final int length;
     private final byte[] bytes;
 
-    public SerializedPrestoSparkPage(byte[] bytes)
+    public PrestoSparkRow(int partition, int length, byte[] bytes)
     {
+        this.partition = partition;
+        this.length = length;
         this.bytes = requireNonNull(bytes, "bytes is null");
     }
 
+    public int getPartition()
+    {
+        return partition;
+    }
+
+    public int getLength()
+    {
+        return length;
+    }
+
     public byte[] getBytes()
     {
         return bytes;
     }
+
+    public long getRetainedSize()
+    {
+        return INSTANCE_SIZE + bytes.length;
```

Comment:
Similarly, we usually use `SizeOf.sizeOf(bytes)`

Reply:
SizeOf is in `Slice` library. I don't want to add it to avoid potential clashes.

Reply:
In this case, should we just copy `sizeOfByteArray` implementation? i.e. something like 

```java
    public long getRetainedSize()
    {
        return INSTANCE_SIZE + sizeOfByteArray(bytes.length);
    }

    // Copied from SizeOf#sizeOfByteArray in Slice library
    private static long sizeOfByteArray(int length)
    {
        return ARRAY_BYTE_BASE_OFFSET + (((long) ARRAY_BYTE_INDEX_SCALE) * length);
    }
```